server/admin_management/api/app/services/validator/idim_validator.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from api.app.integration.idim_proxy import IdimProxyService
+from api.app.constants import (
+    IdimSearchUserParamType,
+    UserType,
+    ERROR_CODE_INVALID_REQUEST_PARAMETER,
+)
+from api.app.schemas import (
+    IdimProxySearchParam,
+    IdimProxyBceidSearchParam,
+    Requester,
+    TargetUser,
+)
+from api.app.utils import utils
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class IdimValidator:
+    def __init__(self, requester: Requester, target_user: TargetUser):
+        self.user_type_code = target_user.user_type_code
+        self.user_name = target_user.user_name
+        self.user_guid = target_user.user_guid
+        self.idim_proxy_service = IdimProxyService(requester)
+
+    def verify_user_exist(self):
+        search_result = None
+        if self.user_type_code == UserType.IDIR:
+            # IDIM web service doesn't support search IDIR by user_guid, so we search by userID
+            search_result = self.idim_proxy_service.search_idir(
+                IdimProxySearchParam(**{"userId": self.user_name})
+            )
+
+            # in edge case, the return guid from search doesn't match the guid given from request parameter
+            # this is unlikely to happen if the request comes from frontend because we also validate user in frontend
+            # but could happen if make backend api call directly
+            if (
+                search_result.get("found")
+                and search_result.get("guid").lower() != self.user_guid.lower()
```

Comment:
Hi Catherine, I probably didn't question guid format myself before, but now I have doubt....
Should the GUID format be case sensitive, so it needs to be exact encoding and not to compare UPPER() or lower()?
I think even IDIM doc isn't clear and even strange is guid from user pool might be case different....

Reply:
yeah, but I feel it's case insensitive, cause I saw both lower and upper cases before, forgot where.